components/case/caseController.js

```diff
@@ -29,6 +30,28 @@ const getCase = async (req, res, next) => {
   }
 };
 
+const allCases = async (req, res, next) => {
+  try {
+    const cases = await caseDao.streamAllCases((stream) => {
+      res.status(200);
+      stream.pipe(JSONStream.stringify()).pipe(res);
+      stream.on('error', (err) => {
+        // log the error and prematurely end the response
```

Comment:
Is there any way to differentiate on the consuming end a prematurely ended stream that hit an error and a fully completed stream?

Reply:
> Is there any way to differentiate on the consuming end a prematurely ended stream that hit an error and a fully completed stream?

The prematurely-ended stream won't pass cleanly through a JSON parser. Is it the case that Shesmu consumes the stream until the host signals it's done? In which case, whether the stream can be deserialized or not determines whether the input is valid.

This is the same implementation used in the `nabu` input format.

Reply:
> Would this solve the issue happening now where test cases are expecting a different exit status?

I don't quite know why the tests sometimes fail like this - `Error: error: relation "fileqc" does not exist` shows it's querying the database but the table doesn't exist. But the table has existed for all the previous tests!